lucetc/src/decls.rs

```diff
@@ -117,18 +117,31 @@ impl<'a> ModuleDecls<'a> {
         for ix in 0..decls.info.functions.len() {
             let func_index = UniqueFuncIndex::new(ix);
 
+            // Get the name for this function from the module names section, if it exists.
+            // Because names have to be unique, we append the index value (ix) to the name.
+            fn custom_name_for<'a>(
+                ix: usize,
+                func_index: UniqueFuncIndex,
+                decls: &mut ModuleDecls<'a>,
+            ) -> Option<String> {
+                decls
+                    .info
+                    .function_names
+                    .get(&func_index)
+                    .cloned()
+                    .map(|s| format!("{}_{}", s, ix))
```

Comment:
:point_up:  this lil' friend here is a helper function for grabbing the name of a function we may have found while parsing. `UniqueFuncIndex` doesn't implement `fmt::Display` so I was using `func_index`, since I don't think we really need to add that derivation while we have a perfectly good `ix`.